webpack/assets/javascripts/react_app/components/Legacy/index.js

```diff
@@ -0,0 +1,47 @@
+import React, { useEffect, useState } from 'react';
+import { useSelector } from 'react-redux';
+import PropTypes from 'prop-types';
+import { isEqual } from 'lodash';
+import { selectReferer } from '../../ReactApp/ReactAppSelectors';
+import DangerouslyInnerHTML from '../InnerHTML';
+import { getLegacyContent } from './helpers';
+
+const LegacyContent = ({ location }) => {
+  const [content, setContent] = useState(null);
+  const [reload, setReload] = useState(false);
+  // The server needs the referer header for redirection
+  const referer = useSelector(selectReferer);
+  const { pathname, search } = location;
+
+  useEffect(() => {
+    /* When a referer exists, no need for full page reload
+       an ajax call occours for getting new legacy content is invoked */
+    if (referer) getLegacyContent(location, setContent, setReload, referer);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [location]);
+
+  useEffect(() => {
+    // triggres a full page reload when needed
+    reload && window.location.assign(`${pathname}${search}`);
+  }, [pathname, search, reload]);
+  return !reload && referer && <DangerouslyInnerHTML html={content} />;
```

Comment:
prefer a blank line here so I don't think the `return` is part of the `useEffect`